app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[edit update destroy]
+  before_action :set_post, only: %i[show]
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    if user_signed_in?
```

Comment:
Move it at `before_action` as suggested up there

Reply:
we fix the issues